### DISCUSSION TITLE
how to convert  args.get_or_undefined(0) to Uint8Array ?

### DISCUSSION THREAD

**Discussion by i18nsite** [boa-dev/boa#3963] [Q&A]
how to convert  args.get_or_undefined(0) to Uint8Array ?

**Comment by dev-abinash-sharma** [comment]
args.get_or_undefined(0) returns a JsValue, so you need to convert it to Uint8Array using js_sys::Uint8Array.

Example:

use js_sys::Uint8Array;
use wasm_bindgen::JsCast;

let value = args.get_or_undefined(0);
let uint8_array: Uint8Array = value.dyn_into().unwrap();

If you need the data as Rust bytes (Vec<u8>):

let bytes: Vec<u8> = uint8_array.to_vec();

Alternatively, you can construct it directly:

let bytes = Uint8Array::new(&args.get_or_undefined(0)).to_vec();

Just ensure that the JavaScript side actually passes a Uint8Array, otherwise the cast will fail.

**Reply by jedel1043** [reply]
We're not `wasm_bindgen`, so this doesn't apply to `boa_engine` 😅

**Comment by jedel1043** [comment]
The correct way to do this is essentially
```rust
use boa_engine::object::builtins::JsUint8Array;
use boa_engine::value::TryFromJs;

let value = args.get_or_undefined(0);
let uint8_array = JsUint8Array::try_from_js(value, context).unwrap()
```
